Modules/syslogmodule.c

```diff
@@ -180,17 +198,8 @@ syslog_syslog(PyObject * self, PyObject * args)
 
     /*  if log is not opened, open it now  */
     if (!S_log_open) {
-        PyObject *openargs;
-
-        /* Continue even if PyTuple_New fails, because openlog(3) is optional.
-         * So, we can still do logging in the unlikely event things are so hosed
-         * that we can't do this tuple.
-         */
-        if ((openargs = PyTuple_New(0))) {
-            PyObject *openlog_ret = syslog_openlog(self, openargs, NULL);
-            Py_XDECREF(openlog_ret);
-            Py_DECREF(openargs);
-        }
+        PyObject *openlog_ret = syslog_openlog_impl(module, NULL, 0, LOG_USER);
```

Comment:
`syslog_openlog_impl()` might set an exception, do you think we should call `PyErr_Clear()` in case it fails in order to swallow it?

Reply:
Good catch! Do you mind to create a separate PR for this (so it can be backported)?

I think that it is better to return NULL rather of swallowing an exception (and we can make `syslog_openlog_impl()` not swallowing *all* exceptions in future).

Reply:
Are you sure returning `NULL` is the right thing? As for now, if `PyTuple_New()` fails we do nothing and call `syslog(3)` anyways (https://github.com/python/cpython/blob/main/Modules/syslogmodule.c#L188).
`openlog(3)` is optional - so why skipping the entire log process?

Reply:
The function should either return NULL and set an exception, or return non-NULL and not set an exception. Violation of this rule leads to mysterious bugs in the following code. In the debug build it leads to a crash.

So you should either clear an exception, or immediately return NULL. But swallowing arbitrary exceptions is in general a bad practice -- it can silence programming or environment errors or cancel user interrupt.

`openlog(3)` is optional, as well as Python's `syslog.openlog()`, but in Python the code tries to determine ident from Python's `sys.argv[0]`, and in C it is determined from C's `argv`. They can be different. This is why the code calls `syslog_openlog()` here instead of just `openlog(3)`.

Reply:
Marking this as resolved, as this was fixed by #95264